### PR TITLE
Trace operator for matrix and matrix3D

### PR DESCRIPTION
**Pull request by kmokstad**
Just some convenience methods. Also (first commit) an extra argument for the SIMbase::project method such that it can be used when we only want a specified component.

### CODE REVIEW

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-poisson=23 please

File: src/LinAlg/matrix.h [hunk 1]
```diff
@@ -916,6 +919,12 @@ namespace utl //! General utility classes and functions.
       return col;
     }
 
+    //! \brief Return the trace of the \a i1'th sub-matrix.
+    T trace(size_t i1) const
+    {
+      return this->elem.sum(i1-1,this->n[0]*(this->n[1]+1));
```
Comment (akva2) [inline src/LinAlg/matrix.h]: a index check is possibly appropriate.

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-poisson=23 please

File: src/LinAlg/matrix.h (hunk 1 above)
Comment (kmokstad) [reply src/LinAlg/matrix.h]: Yes, indeed. Added.

**Comment by kmokstad** [comment]
jenkins build this with downstreams ifem-poisson=23 please

**Pull request merged**